### PR TITLE
[Leaflet] Making an L.Control class that extends L.Class so you can extend Control

### PR DESCRIPTION
**Pull request by reblace**
…xtend Control class

Please fill in this template.

- [X] Make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [X] Run `tsc` without errors.
- [NA] Run `npm run lint package-name` if a `tslint.json` is present.

The change is to allow the type of thing I added to the tests. Specifically, you can now create a new class that extends L.Control. L.Class exposes several static methods, and these static methods do exist on L.Control, but since there was no L.Control class (only an L.Control interface), these static methods were not available.

Ideally, we should convert most of the remaining interfaces to classes so we can merge the definitions and you can extend all of them. Layer is already specified this way. Let me know if you agree and I can put together a more extensive PR that handles that refactor. But since this is a small change, I wanted to keep it isolated.

### CODE REVIEW

**Comment by sheetalkamat** [comment]
@reblace Thank you for your contribution. 
Can you please fix the failing Travis CI tests and update the PR!

**Comment by reblace** [comment]
I'm not sure how this failure wasn't already present. The failure is in the leaflet-geocoder-mapzen leaflet plugin. I found the source of the duplicate, but it's in the 0.7.29 tag of @types/leaflet. I'm not sure how this failure is new in this PR or why it's not already failing on master. If you look at the master build log, it isn't actually running any tests. I can fix it, but it looks to be completely unrelated to this PR (AFAICT).

**Comment by sheetalkamat** [comment]
@andy-ms please take a look at the tests

**Comment by andy-ms** [comment]
Not sure why it passed tests last time, but I think you can just remove the duplicate `type PositionString` in `leaflet-geocoder-mapzen` (along with the TODO comment).
`leaflet-geocoder-mapzen` currently compiles fine in `typescript@next` but not in `typescript@2.0`, so that might have something to do with it.

**Comment by reblace** [comment]
Ok, I'll fix it and update the PR

**Comment by reblace** [comment]
This should address the issue. The leaflet-geocoder-mapzen plugin references leaflet >0.7, and a recent update added the ```PositionString``` type, which was colliding.

File: leaflet/index.d.ts [hunk 1]
```diff
@@ -962,6 +962,9 @@ declare namespace L {
         position?: ControlPosition;
     }
 
+    export class Control extends Class {
```
Comment (andy-ms) [inline leaflet/index.d.ts]: You don't need both a class and interface in a case like this. Just move the body of the interface inside the class.

**Comment by reblace** [comment]
Fixed conflict with mapzen and moved Control interface into Control class.

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**